CMakeLists.txt

```diff
@@ -459,6 +459,12 @@ try_append_cxx_flags("-fstack-reuse=none" TARGET core_interface)
 if(ENABLE_HARDENING)
   add_library(hardening_interface INTERFACE)
   target_link_libraries(core_interface INTERFACE hardening_interface)
+
+  target_compile_options(hardening_interface INTERFACE
```

Comment:
Maybe we should add this to depends instead?

Reply:
By default hardening is enabled in the CMake config and the depends build will benefit from that. Only if depends build uses `make NO_HARDEN=1` then it will disable the hardening in CMake.